### PR TITLE
Do not assign non-existing ids for annotations.

### PR DESCRIPTION
**Pull request by Integral**
Currently when you pasting annotated text (let's say HTML one) only first paragraph got annotated.
That's happening typically because you are assigning non existing ids to annotations (at least when you are pasting content inside empty document).

### CODE REVIEW

**Comment by obuchtala** [comment]
Would be good to have a test case revealing the problem.

**Comment by Integral** [comment]
When a lot of red is a good sign :)
I will look closely and try to make it green, #1083 is here and other things also I believe...

**Comment by obuchtala** [comment]
Haha. It seems you really broke it ;)

**Comment by obuchtala** [comment]
Let me know if I can help :)

**Comment by obuchtala** [comment]
Oh, I see... you added a revealing test.
Very good!
Let's fix it. Let me know if I can help you.

**Comment by Integral** [comment]
I will do it today ;)

File: ui/ClipboardImporter.js [hunk 1]
```diff
@@ -90,8 +90,37 @@ class ClipboardImporter extends HTMLImporter {
     // specific format, e.g., id="docs-internal-guid-5bea85da-43dc-fb06-e327-00c1c6576cf7"
     let bold = body.find('b')
     if (bold && /^docs-internal/.exec(bold.id)) {
-      return bold
+      body = bold
     }
+
+    body.findAll('span').forEach(span => {
+      const style = span.el.style
```
Comment (obuchtala) [inline ui/ClipboardImporter.js]: In general you should use span.getNativeElement() instead of span.el as  does not work in the memory DOM.

Comment (obuchtala) [reply ui/ClipboardImporter.js]: ... but as you see below, accessing the native element here is not a good idea

File: ui/ClipboardImporter.js [hunk 2]
```diff
@@ -90,8 +90,37 @@ class ClipboardImporter extends HTMLImporter {
...
+      // Google Docs uses spans with inline styles 
+      // insted of inline nodes
+      // We are scanning each span for certain inline styles:
+      // font-weight: 700 -> <b>
+      // font-style: italic -> <i>
+      // vertical-align: super -> <sup>
+      // vertical-align: sub -> <sub>
+      let nodeType
+      if(style['font-weight'] === '700') {
```
Comment (obuchtala) [inline ui/ClipboardImporter.js]: You should use getStyle('font-weight') here. This works in browser and nodejs.

File: ui/ClipboardImporter.js [hunk 3]
```diff
@@ -90,8 +90,37 @@ class ClipboardImporter extends HTMLImporter {
...
+        nodeType = 'b'
+      } else if (style['font-style'] === 'italic') {
```
Comment (obuchtala) [inline ui/ClipboardImporter.js]: same same

File: ui/ClipboardImporter.js [hunk 4]
```diff
@@ -90,8 +90,37 @@ class ClipboardImporter extends HTMLImporter {
...
+        nodeType = 'i'
+      } else if (style['vertical-align'] === 'super') {
```
Comment (obuchtala) [inline ui/ClipboardImporter.js]: span.getStyle('vertical-align')

File: ui/ClipboardImporter.js [hunk 5]
```diff
@@ -90,8 +90,37 @@ class ClipboardImporter extends HTMLImporter {
...
+        nodeType = 'sup'
+      } else if (style['vertical-align'] === 'sub') {
```
Comment (obuchtala) [inline ui/ClipboardImporter.js]: dito

**Comment by obuchtala** [comment]
I have the feeling that it is very hard to support these combined formatting annotations.
Maybe we accept for now, that there will be more fragments than necessary, and relax the test-case a bit.
On the long run we should try to improve the result by fusing adjacent annotations of the same type.

**Comment by Integral** [comment]
@oliver---- there are new problems. And the main one is that we have same annotation type overlapping. Is that ok generally that we have two strongs on some range?

**Comment by obuchtala** [comment]
It is ok I think. In general we can not avoid this, so we need make sure to be able to deal with it.

**Comment by obuchtala** [comment]
@michael is having two bolds at the same place a problem for the UI?
As I said, this can actually come in from XML/HTML, and is valid content there.

**Comment by Integral** [comment]
God! Merge it pls while it green!
We definitely need to tidy html in future. That's the only independent from editor way...

**Comment by obuchtala** [comment]
Hehe :)
I'll take another look first, and then ...

**Comment by obuchtala** [comment]
👍

**Comment by obuchtala** [comment]
@Integral Good work. Thx!

**Pull request merged**